### ISSUE TITLE
inspect_app() is not the only possible source of a bundle identifier

### ISSUE THREAD

**Issue by homebysix** [homebysix/recipe-robot#40]
At the moment, Recipe Robot doesn't build a pkg recipe if there's no bundle_id in facts. However, I don't think we've considered that there are other possible ways to determine the bundle identifier even if we weren't able to collect it during `inspect_app()`.

- Many Sparkle feeds provide bundle identifiers that `SparkleUpdateInfoProvider` grabs.
- `AppDmgVersioner` can be used to determine a bundle identifier. As long as we include that before the `PkgCreator` processor, we should be good to go.

**Comment by sheagcraig** [comment]
So let me get this straight-
inspect_app is happy if it has the app already on the system, where it's going to definitely need a bundle ID, and thus, inspect_app can 100% of the time get the ID.

I hear you about `SparkleUpdateInfoProvider`, although I would think that the ID in the actual app should be considered canonical, even over anything provided as Input variables or overrides.

I think it's very safe to assume that an app, if downloaded, will have a bundle ID, so yes, you can build a package recipe, knowing that it's going to need _something_ to get the bundle ID (and version).

What that something is, is tricky. I think AppDmgVersioner right before PkgCreator is a good candidate, as it should be able to handle ANYTHING that is going to be immediately preceding PkgCreator. Right?

**Comment by homebysix** [comment]
Yes, if we've run `inspect_app()` then we should be pretty confident that we have the correct bundle id (and yes, that bundle id should override anything stored before it; I'll make that change).

If we haven't successfully inspected an app at some point along the line, it's likely that RR isn't going to be able to produce a full recipe set without manual intervention. But "likely" ≠ "guaranteed."

I'll try to find some edge case examples.

**Comment by homebysix** [comment]
Recipe Robot has a new junior developer named Claude, who adds the following.

---

## Current State

This issue remains **valid and unresolved** as of the current codebase.

## Analysis

**Problem:** Recipe Robot skips pkg recipe generation when `bundle_id` is not present in facts, even when other processors could provide the bundle identifier at runtime.

**Evidence:**
- `recipe_generator.py` lines 945-955 contain the blocking logic with TODO comment referencing this issue (#40)
- The code explicitly skips pkg recipe creation if `"bundle_id" not in facts`

**Available Solution:**
- `AppDmgVersioner` processor outputs `bundleid` variable containing the bundle identifier
- Recipe Robot already uses `AppDmgVersioner` in download recipes (lines 717-722 in `recipe_generator.py`)
- Pkg recipes could reference `%bundleid%` instead of requiring `bundle_id` in facts

**Impact:**
This limitation prevents pkg recipe generation for apps distributed as DMGs where the bundle identifier could be extracted by `AppDmgVersioner` during recipe execution.

## Note on Sparkle Feeds

Investigation shows that `SparkleUpdateInfoProvider` does not typically provide bundle identifiers - it only outputs `url`, `version`, and `additional_pkginfo`.
